### PR TITLE
Changes after Fuzz Testing useFmtOnly

### PR DESCRIPTION
**Pull request by rene-ye**
Fuzzing caught a few unexpected errors. Most changes are just checking Iterator.hasNext() before moving the iterator.

### CODE REVIEW

File: src/main/java/com/microsoft/sqlserver/jdbc/SQLServerFMTQuery.java [hunk 1]
```diff
@@ -111,3 +115,19 @@ String getFMTQuery() {
...
+
+class SQLServerErrorListener extends BaseErrorListener {
+    static final private java.util.logging.Logger logger = java.util.logging.Logger
+            .getLogger("com.microsoft.sqlserver.jdbc.internals.SQLServerFMTQuery");
+
+    @Override
+    public void syntaxError(Recognizer<?, ?> recognizer, Object offendingSymbol, int line, int charPositionInLine,
+            String msg, RecognitionException e) {
+        if (logger.isLoggable(java.util.logging.Level.FINE)) {
+            logger.fine("Error occured during token parsing: " + msg);
```
Comment (lilgreenbird) [inline src/main/java/com/microsoft/sqlserver/jdbc/SQLServerFMTQuery.java]: why not just 1 call?

Comment (rene-ye) [reply src/main/java/com/microsoft/sqlserver/jdbc/SQLServerFMTQuery.java]: Wanted 2 lines. 1 line for the error message and the 2nd line for what token couldn't be read and where it is.

Comment (lilgreenbird) [reply src/main/java/com/microsoft/sqlserver/jdbc/SQLServerFMTQuery.java]: "\n"?

Comment (rene-ye) [reply src/main/java/com/microsoft/sqlserver/jdbc/SQLServerFMTQuery.java]: Think it shows up a little weird in logs, every line generally has some prefix with timestamp info, etc. There are other places in the driver that double logs like this.

File: src/main/java/com/microsoft/sqlserver/jdbc/SQLServerParser.java [hunk 2]
```diff
@@ -201,7 +213,7 @@ private static String findColumnAfterParameter(SQLServerTokenIterator iter) {
         StringBuilder sb = new StringBuilder();
         while (sb.length() == 0 && iter.hasNext()) {
```
Comment (lilgreenbird) [inline src/main/java/com/microsoft/sqlserver/jdbc/SQLServerParser.java]: 0 == sb.length(), multiple occurrences, search for "== 0"

File: src/main/java/com/microsoft/sqlserver/jdbc/SQLServerFMTQuery.java (hunk 1 above)
Comment (lilgreenbird) [reply src/main/java/com/microsoft/sqlserver/jdbc/SQLServerFMTQuery.java]: okey, no big deal I guess...

**Comment by ulvii** [comment]
Format the files.

File: src/main/java/com/microsoft/sqlserver/jdbc/SQLServerResource.java [hunk 3]
```diff
@@ -590,5 +590,8 @@ static String getResource(String key) {
                     "Invalid syntax: OPENQUERY/OPENJSON/OPENDATASOURCE/OPENROWSET/OPENXML must be preceded by round brackets"},
             {"R_invalidCTEFormat",
                     "Invalid syntax: AS must be followed by round brackets in Common Table Expressions."},
-            {"R_noTokensFoundInUserQuery", "Invalid query: No tokens were parsed from the SQL provided."}};
+            {"R_noTokensFoundInUserQuery", "Invalid query: No tokens were parsed from the SQL provided."},
+            {"R_invalidUserSQL", "An error occured when attempting to parse user SQL. Please verify SQL syntax."},
+            {"R_invalidInsertValuesQuery", "Error when matching VALUES list to Table columns. Please verify user SQL."},
```
Comment (ulvii) [inline src/main/java/com/microsoft/sqlserver/jdbc/SQLServerResource.java]:
```suggestion
            {"R_invalidInsertValuesQuery", "Error when matching VALUES list to table columns. Please verify SQL syntax."},
```

File: src/main/java/com/microsoft/sqlserver/jdbc/SQLServerResource.java [hunk 4]
```diff
@@ -590,5 +590,8 @@ static String getResource(String key) {
...
+            {"R_invalidValuesList", "Error when reading VALUES list. Please verify user SQL."}};
```
Comment (ulvii) [inline src/main/java/com/microsoft/sqlserver/jdbc/SQLServerResource.java]:
```suggestion
            {"R_invalidValuesList", "Error when reading VALUES list. Please verify SQL syntax."}};
```

File: src/main/java/com/microsoft/sqlserver/jdbc/SQLServerResource.java [hunk 5]
```diff
@@ -590,5 +590,8 @@ static String getResource(String key) {
                     "Invalid syntax: OPENQUERY/OPENJSON/OPENDATASOURCE/OPENROWSET/OPENXML must be preceded by round brackets"},
             {"R_invalidCTEFormat",
                     "Invalid syntax: AS must be followed by round brackets in Common Table Expressions."},
-            {"R_noTokensFoundInUserQuery", "Invalid query: No tokens were parsed from the SQL provided."}};
+            {"R_noTokensFoundInUserQuery", "Invalid query: No tokens were parsed from the SQL provided."},
+            {"R_invalidUserSQL", "An error occured when attempting to parse user SQL. Please verify SQL syntax."},
```
Comment (ulvii) [inline src/main/java/com/microsoft/sqlserver/jdbc/SQLServerResource.java]:
```suggestion
            {"R_invalidUserSQL", "An error occurred when attempting to parse user SQL. Please verify SQL syntax."},
```

**Pull request merged**